libvast/src/table_slice.cpp

```diff
@@ -153,12 +153,20 @@ table_slice::table_slice(chunk_ptr&& chunk, enum verify verify) noexcept
       },
       [&](const auto& encoded) noexcept {
         auto& state_ptr = state(encoded, state_);
+        // We create a second chunk with an intentionally decoupled reference
+        // count here that we bind to the lifetime of the original chunk. This
+        // avoids cyclic references between the table slice and its
+        // encoding-specific state.
+        auto decoupled_chunk = chunk::make(as_bytes(*chunk_), []() noexcept {});
```

Comment:
Does this warrant it's own `chunk::decouple()` function?

Reply:
I don't think we want to expose that API, it's so easy to misuse. I'd rather not use it here and refactor the table slice dispatch logic to be easier to use instead.